include/aarch64/gpio.h

```diff
@@ -0,0 +1,12 @@
+#ifndef _GPIO_H_
+#define _GPIO_H_
+
+struct resource_t;
+
+void delay(int64_t count);
+
+void gpio_function_select(resource_t *r, unsigned pin, unsigned func);
```

Comment:
Perhaps a short docstring would be helpful for readers. Could you link [Raspberry Pi Pinout](https://pinout.xyz) here for reference?

BTW. What is maximum value for `pin`. Should we check that with `assert`?

Reply:
I added assert and links to documentation.